databricks/koalas/window.py

```diff
@@ -42,6 +60,91 @@ def __getattr__(self, item: str) -> Any:
                 return partial(property_or_func, self)
         raise AttributeError(item)
 
+    def _apply_as_series_or_frame(self, func):
+        """
+        Decorator that can wraps a function that handles Spark column in order
+        to support it in both Koalas Series and DataFrame.
+        Note that the given `func` name should be same as the API's method name.
+        """
+        from databricks.koalas import DataFrame, Series
+
+        if isinstance(self.kdf_or_kser, Series):
+            kser = self.kdf_or_kser
+            return kser._with_new_scol(
+                func(kser._scol, kser._internal.index_scols)).rename(kser.name)
+        elif isinstance(self.kdf_or_kser, DataFrame):
+            kdf = self.kdf_or_kser
+            applied = []
+            for column in kdf.columns:
+                applied.append(
+                    getattr(kdf[column].rolling(self._window_val + 1,
+                            self._min_periods), func.__name__)())
+
+            sdf = kdf._sdf.select(
+                kdf._internal.index_scols + [c._scol for c in applied])
+            internal = kdf._internal.copy(
+                sdf=sdf,
+                data_columns=[c._internal.data_columns[0] for c in applied],
+                column_index=[c._internal.column_index[0] for c in applied])
+            return DataFrame(internal)
+
+    def count(self):
+        """
+        The rolling count of any non-NaN observations inside the window.
+
+        .. note:: the current implementation of this API uses Spark's Window without
+            specifying partition specification. This leads to move all data into
+            single partition in single machine and could cause serious
+            performance degradation. Avoid this method against very large dataset.
+
+        Returns
+        -------
+        Series or DataFrame
+            Returned object type is determined by the caller of the rolling
+            calculation.
+
+        See Also
+        --------
+        Series.rolling : Calling object with Series data.
+        DataFrame.rolling : Calling object with DataFrames.
+        DataFrame.count : Count of the full DataFrame.
+
+        Examples
+        --------
+        >>> s = ks.Series([2, 3, float("nan"), 10])
+        >>> s.rolling(1).count()
+        0    1.0
+        1    1.0
+        2    0.0
+        3    1.0
+        Name: 0, dtype: float64
+
+        >>> s.rolling(3).count()
+        0    1.0
+        1    2.0
+        2    2.0
+        3    2.0
+        Name: 0, dtype: float64
+
+        >>> s.to_frame().rolling(1).count()
+             0
+        0  1.0
+        1  1.0
+        2  0.0
+        3  1.0
+
+        >>> s.to_frame().rolling(3).count()
+             0
+        0  1.0
+        1  2.0
+        2  2.0
+        3  2.0
+        """
+        def count(scol, _):
```

Comment:
why `_` is needed?

Reply:
Because in the Decorator `_apply_as_series_or_frame`, Rolling functions are called `func(ker._scol, ker._internal.index_scols)` like below.


<img width="646" alt="스크린샷 2019-11-05 오후 12 51 18" src="https://user-images.githubusercontent.com/44108233/68177265-19ee4380-ffcb-11e9-8989-27d4d0f317ab.png">


But only for the case of `count`, we don't need `ker._internal.index_scols` parameter.

Reply:
i fixed them!